### PR TITLE
[R4R]Add docs about bsc and cross chain between bc and bsc

### PR DESCRIPTION
**Pull request by guagualvcha**

### CODE REVIEW

File: docs/faq/bsc/bsc.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+* Native cross-chain trasfer & communication: Binance DEX remains a liquid venue of exchange of assets on Binance Chain and Binance Smart Chain"
+
+## What are the benefits for developers to build on Binance chain?
+
+Binance Chain opens the gate for users to take advantage of the fast transferring and trading
+
+## There are already many wallets/tools support Binance chain
+
+There are already 110 assets on Binance Chain
+The introduction of BRP8 will be an innovative way for tokenization of properties
```
Comment (guagualvcha) [inline docs/faq/bsc/bsc.md:34]: `BRP8 ` -> `BEP8 `

File: docs/faq/bsc/bsc.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+* EVM-compatible: Binance Smart Chain supports all the existing Ethereum tooling
+Fast block time, cheaper cost
+* Native cross-chain trasfer & communication: Binance DEX remains a liquid venue of exchange of assets on Binance Chain and Binance Smart Chain"
+
+## What are the benefits for developers to build on Binance chain?
+
+Binance Chain opens the gate for users to take advantage of the fast transferring and trading
+
+## There are already many wallets/tools support Binance chain
```
Comment (guagualvcha) [inline docs/faq/bsc/bsc.md:31]: This looks like not a question?

File: docs/faq/bsc/bsc.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+There are already 110 assets on Binance Chain
+The introduction of BRP8 will be an innovative way for tokenization of properties
+
+## What make Binance Smart Chain different?
+
+Key Innovations:
```
Comment (guagualvcha) [inline docs/faq/bsc/bsc.md:38]: Here is a todo I guess.

File: docs/faq/bsc/cross.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Cross-chain Communication
+
+## How much is cross-chain transfer fee?
+
+The total cost of transfer from BC to BSC is composed of 3 parts:
+
+* Fee for `bnbcli bridge transfer-out` 0.01BNB,  pay validators on Binance Chain
```
Comment (guagualvcha) [inline docs/faq/bsc/cross.md:7]: Not a good idea to use cli command to describe the fee.

File: docs/faq/bsc/cross.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+* Fee for BSC-relayers 0.01BNB
+
+* Call TokenHub Contract: You need to pay BNB for calling smart-contract on BSC, this transaction is metered by gas fee, which is a global parameter. At the moment, you need to pay about 0.001BNB.
```
Comment (guagualvcha) [inline docs/faq/bsc/cross.md:11]: No need to mention about gas fee.

File: docs/faq/bsc/cross.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+The total cost of transfer from BSC to BC is composed of 2 parts:
+
+* Fee for Oracle-relayers 0.02BNB, pay for BSC relayers
```
Comment (guagualvcha) [inline docs/faq/bsc/cross.md:15]: No such fee, please confirm

File: docs/faq/bsc/cross.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+* Call TokenHub Contract: You need to pay BNB for calling smart-contract on BSC, this transaction is metered by gas fee, which is a global parameter. At the moment, you need to pay about 0.001BNB.
+
+## What's is an Oracle relayer?
+
+Relayer transport packages of Block Header and Merkle Proof verification to let BC validators verify the block transactions and even the state value .
```
Comment (guagualvcha) [inline docs/faq/bsc/cross.md:21]: The description is not correct. May confirm with Zhenxin again.

File: docs/faq/bsc/cross.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+* Call TokenHub Contract: You need to pay BNB for calling smart-contract on BSC, this transaction is metered by gas fee, which is a global parameter. At the moment, you need to pay about 0.001BNB.
+
+The total cost of transfer from BSC to BC is composed of 2 parts:
+
+* Fee for Oracle-relayers 0.02BNB, pay for BSC relayers
+
+* Call TokenHub Contract: You need to pay BNB for calling smart-contract on BSC, this transaction is metered by gas fee, which is a global parameter. At the moment, you need to pay about 0.001BNB.
+
+## What's is an Oracle relayer?
```
Comment (guagualvcha) [inline docs/faq/bsc/cross.md:19]: I think you want to write Bsc relayer?

File: docs/faq/bsc/cross.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+Relayer transport packages of Block Header and Merkle Proof verification to let BC validators verify the block transactions and even the state value .
+
+## What's an oracle?
+
+The execution of the transanction wil emit Events, and such events can be observed and packaged in certain “Oracle” onto BC.
```
Comment (guagualvcha) [inline docs/faq/bsc/cross.md:25]: Who can be oracle relayer? and when an event become true on BC?

File: docs/faq/bsc/cross.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+## Which wallet support cross-chain transfer?
+
+You need to use [MyEtherWallet]() to call contracts and use `bnbcli` for complementry commands
```
Comment (guagualvcha) [inline docs/faq/bsc/cross.md:29]: `bnbcli ` or `tbnbcli `

File: docs/faq/bsc/token.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
+# Tokens in Binance Chain and Binance Smart Chain
+
+## What role does BNB play in securing Binance Smart Chain?
+
+BNB is the token used to prevent `nothing-at-stake` problem.
+
+## What are tokens on Binance Smart Chain?
+
+BNB is the native token for BSC, and pegged BEP2 tokens on BSC is called BSC.
```
Comment (guagualvcha) [inline docs/faq/bsc/token.md:9]: ` is called BSC`  ==>` is called BEP2E.` ?

File: docs/faq/bsc/token.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+## Where can I find out more about staking economics and rewards?
+
+You can read details in White Paper: <https://github.com/binance-chain/whitepaper/blob/master/WHITEPAPER.md#rewarding>
+
+## Where do network rewards come from?
+
+Validators and delegators will earn rewards from the following sources:
+
+* Cross-chain transfers
```
Comment (guagualvcha) [inline docs/faq/bsc/token.md:39]: `Cross-chain transfers`  is not one of the rewards comes.

File: docs/faq/bsc/val.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+slash amount, cross-chain transfer fees
+
+## How to join testnet as a validator?
+
+1. Chooe your own server/PC
+2. Install software:
+3. Create a wallet and get some BNB
+4. Run your fullnode and keep it synced
+5. Stake your BNB on BC, the top 21 most staked nodes to be the validator set"
```
Comment (guagualvcha) [inline docs/faq/bsc/val.md:21]: delete  `"`

File: docs/faq/bsc/val.md [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+## What are hardware requirements of running a validator node?
+
+Processing transactions is mostly CPU bound. Therefore we recommend running CPU optimized servers.
+Directly facing internet (public IP, no NAT)
+16 cores CPU
```
Comment (guagualvcha) [inline docs/faq/bsc/val.md:35]: `16 cores` is too much... I hope we can have a suggestion requirements and least requirements.

File: docs/smart-chain/guides/concepts/consensus.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+
+1. With quite a few blocks to confirm(should be less than Ethereum), better no fork in most cases.
+2. Blocking time should be faster than Ethereum, i.e. 5 seconds or less.
+3. No inflation, the block reward is transactiongas fees.
+4. As much as compatible as Ethereum.
+5. With governance as powerful as cosmos.
+
+[Geth](https://github.com/ethereum/go-ethereum/wiki/geth) implements two kinds of consensus engine: ethash(based on PoW) and [clique](https://ethereum-magicians.org/t/eip-225-clique-proof-of-authority-consensus-protocol/1853)(base on PoA). Ethash is obviously not a good option for BSC because of lacking hash power on BSC. Clique has smaller blocking time and is invulnerable to 51% attack while doing as little to the core data structure as possible to preserve existing Ethereum client compatibility,  it seems to be a good choice except for its lack of powerful staking and governance capability on-chainon chain.  On the other hand, the Binance chain is built on Cosmos which does have a perfect staking and governance mechanism. Naturally, we try to propose a consensus engine that:
```
Comment (guagualvcha) [inline docs/smart-chain/guides/concepts/consensus.md:12]: `capability on-chainon chain.`?

File: docs/smart-chain/guides/concepts/consensus.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+Binance chain does the staking and governance parts for BSC.
+ValidatorSet change of BSC is updated through interchain communication.
+Consensus engine of BSC keeps as simple as clique.
+
+Speaking of popular implementations of PoA consensus, it turns out that Bor follows athe similar design as PoSA. However, Bor still have some deficiencies, the critical one is that its security and reliability depends on running a trustful and stable Heimdall node.  We will try to borrow a few parts from Bor and   propose a new consensus engine to achieve all these goals.
```
Comment (guagualvcha) [inline docs/smart-chain/guides/concepts/consensus.md:17]: Please delete line 17, we are invastigating Bor when writting the doc, we may not menthion the bad aspect of other chain.

File: docs/smart-chain/guides/concepts/consensus.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+### How to Validate/Replay a block
+
+#### Step1: VerifyHeader
+Verify the block header when receiving a new block.
+
+* Verify the signature of the coinbase is in `extraData` of the `blockheader`
+* Compare the block time of the `blockHeader` and the expected block time that the signer suppose to use, will deny a `blockerHeader` that is smaller than expected. It helps to prevent a selfish validator from rushing to seal a block.
+* The `coinbase` should be the signer and the difficulty should be expected value.
+
+Besides the above change, the logic of distributing gas fee also need change in miner layer, refer to Reward Distribution section.
```
Comment (guagualvcha) [inline docs/smart-chain/guides/concepts/consensus.md:55]: please delete line 55. It is a implement detail

File: docs/smart-chain/guides/concepts/system-contract.md [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+* function **getAppHash**(uint64 height)
+getAppHash provides a method to get the verified appHash at the specified height. Besides, If the header the specified height have not be verified, then zero value will be returned.
+
+#### Merkle Proof Verification Library
+
+* function **verifyMerkleProof**(int64 height, byte[] key, byte[] value, byte[] proof) bool
+
+verifyMerkleProof reassembles user parameters and call the the above precompiled contract to validate the proof. Any contract which need to verify merkle proof just need to import this Library.
+
+## Other Build-in System Contract
```
Comment (guagualvcha) [inline docs/smart-chain/guides/concepts/system-contract.md:57]: I will add more about this part.

File: docs/smart-chain/guides/concepts/bsc-relayer.md [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+Relayers for BC-to-BSC communication referred to as **BSC Relayers** are a standalone process that can be run by anyone, and anywhere, except that Relayers must register themselves onto BSC and deposit a certain amount of BNB. Only relaying requests from the registered Relayers will be accepted by BSC.
+
+## Monitor and Parse IBC Event
+As a BSC relayer, it must have proper configurations on the following three items:
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+|    srcIBCChainID  |   uint16   |      IBCChainID of BC, the value is 1 for testnet       |
+|destIBCChainID|uint16|IBCChainID of BSC, the value is 2 for testnet|
+|destChainName|string|name of BSC, the value is “bsc”|
```
Comment (HaoyangLiu) [inline docs/smart-chain/guides/concepts/bsc-relayer.md:13]: > name of BSC, the value is “bsc”

name of targe chain, for Binance Smart Chain, the value should be “bsc”

File: docs/smart-chain/guides/concepts/bsc-relayer.md [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+2. Call MarshalBinaryLengthPrefixed to encode header struct:
+```golang
+func (h *Header) EncodeHeader() ([]byte, error) {
+		bz, err := cdc.MarshalBinaryLengthPrefixed(h)
+		if err != nil {
+			return nil, err
+			}
+		return bz, nil
+	}
+```
```
Comment (HaoyangLiu) [inline docs/smart-chain/guides/concepts/bsc-relayer.md:70]: Refactor indent

File: docs/smart-chain/guides/concepts/bsc-relayer.md [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+## Monitor and Parse IBC Event
+As a BSC relayer, it must have proper configurations on the following three items:
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+|    srcIBCChainID  |   uint16   |      IBCChainID of BC, the value is 1 for testnet       |
+|destIBCChainID|uint16|IBCChainID of BSC, the value is 2 for testnet|
+|destChainName|string|name of BSC, the value is “bsc”|
+
+BSC relayer should try get all block result and pick out the event whose event type is “IBCPackage” from endBlock event table. Mark the block height as **H**. If there is no ibc event, then just skip this block. This is an example of ibc event.
```
Comment (HaoyangLiu) [inline docs/smart-chain/guides/concepts/bsc-relayer.md:15]: > BSC relayer should try get all block result and pick out the event whose event type is “IBCPackage” from endBlock event table.

A BSC relayer is required to parse all block results and pick out all events with event type “IBCPackage” from endBlock event table.

File: docs/smart-chain/guides/concepts/bsc-relayer.md [hunk 22]
```diff
@@ -0,0 +1,113 @@
...
+### Query BC Header at Height
+```golang
+type Header struct {
+  tmtypes.SignedHeader
+  ValidatorSet     *tmtypes.ValidatorSet `json:"validator_set"`
+  NextValidatorSet *tmtypes.ValidatorSet `json:"next_validator_set"`
+}
+```
+
+Call the following rpc methods to build the above Header object:
```
Comment (HaoyangLiu) [inline docs/smart-chain/guides/concepts/bsc-relayer.md:50]: > Call the following rpc methods to build the above Header object:

If an IBCPackage event is found at height **H**, then call the following rpc methods to build the above Header object:

File: docs/smart-chain/guides/concepts/bsc-relayer.md [hunk 23]
```diff
@@ -0,0 +1,113 @@
...
+
+## Query BC Header and IBC Store
+### Query BC Header at Height
+```golang
+type Header struct {
+  tmtypes.SignedHeader
+  ValidatorSet     *tmtypes.ValidatorSet `json:"validator_set"`
+  NextValidatorSet *tmtypes.ValidatorSet `json:"next_validator_set"`
+}
+```
```
Comment (HaoyangLiu) [inline docs/smart-chain/guides/concepts/bsc-relayer.md:48]:
```golang
type Header struct {
  tmtypes.SignedHeader
  ValidatorSet     *tmtypes.ValidatorSet `json:"validator_set"`
  NextValidatorSet *tmtypes.ValidatorSet `json:"next_validator_set"`
}
```

=>

```golang
import tmtypes "github.com/tendermint/tendermint/types"
type Header struct {
  tmtypes.SignedHeader
  ValidatorSet     *tmtypes.ValidatorSet `json:"validator_set"`
  NextValidatorSet *tmtypes.ValidatorSet `json:"next_validator_set"`
}
```

File: docs/smart-chain/guides/concepts/bsc-relayer.md (hunk 20 above)
Comment (HaoyangLiu) [inline docs/smart-chain/guides/concepts/bsc-relayer.md:70]: Encode the Header object to a byte array:

1. Import cdc from tendermint crypto:https://github.com/tendermint/tendermint/blob/master/crypto/encoding/amino/amino.go
2. Call MarshalBinaryLengthPrefixed to encode header struct:
```golang
func (h *Header) EncodeHeader() ([]byte, error) {
		bz, err := cdc.MarshalBinaryLengthPrefixed(h)
		if err != nil {
			return nil, err
			}
		return bz, nil
	}
```

**Change to this:**

Encode the Header object to a byte array:

1. Add dependency on [tendermint v0.32.3](https://github.com/tendermint/tendermint/tree/v0.32.3):
2. Example golang code to encode header struct:
```golang

import (
  amino "github.com/tendermint/go-amino"
  tmtypes "github.com/tendermint/tendermint/types"
)

var cdc = amino.NewCodec()

func init() {
  tmtypes.RegisterBlockAmino(cdc)
}

func (h *Header) EncodeHeader() ([]byte, error) {
  bz, err := cdc.MarshalBinaryLengthPrefixed(h)
  if err != nil {
     return nil, err
  }
  return bz, nil
}

```

File: docs/smart-chain/guides/concepts/bsc-relayer.md [hunk 24]
```diff
@@ -0,0 +1,113 @@
...
+
+Call the following rpc methods to build the above Header object:
+
+|Name|Method|
+| ---- | ---- |
+|tmtypes.SignedHeader|{rpc}/commit?height=**H**|
+|ValidatorSet|{rpc}/validators?height=**H**|
+|NextValidatorSet|{rpc}/validators?height=**H+1**|
+
```
Comment (HaoyangLiu) [inline docs/smart-chain/guides/concepts/bsc-relayer.md:57]: > |tmtypes.SignedHeader|{rpc}/commit?height=**H**|
> |ValidatorSet|{rpc}/validators?height=**H**|
> |NextValidatorSet|{rpc}/validators?height=**H+1**|

|tmtypes.SignedHeader|{rpc}/commit?height=**H+1**|
|ValidatorSet|{rpc}/validators?height=**H+1**|
|NextValidatorSet|{rpc}/validators?height=**H+2**|

File: docs/smart-chain/guides/concepts/bsc-relayer.md [hunk 25]
```diff
@@ -0,0 +1,113 @@
...
+
+|Parameter Name|Type|Value|
+| ---- | ---- | ----------- |
+|msgBytes|[]byte|package bytes|
+|proof|[]byte|merkle proof bytes|
+|height|uint64|**H + 1**|
+|packageSequence|uint64|sequence from attribution value|
+
+## Incentives Mechanism
```
Comment (HaoyangLiu) [inline docs/smart-chain/guides/concepts/bsc-relayer.md:113]: [BSC Relayer Incentive Mechanism](incentives.md)

File: docs/faq/bsc/bsc.md (hunk 1 above)
Comment (huangsuyu) [reply docs/faq/bsc/bsc.md:34]: fixed!

File: docs/faq/bsc/bsc.md (hunk 2 above)
Comment (huangsuyu) [reply docs/faq/bsc/bsc.md:31]: it's updated

File: docs/guides/concepts/bc-bridge.md [hunk 26]
```diff
@@ -0,0 +1,130 @@
...
+| --amount            | 1000000000                                 | amount of tokens to bind                             | Yes          |
+| --contract-address  | 0x6aade9709155a8386c63c1d2e5939525b960b4e7 | contract address of token  in smart chain            | Yes          |
+| --contract-decimals | 18                                         | decimals of token in  smart chain                    | Yes          |
+| --expire-time       | 1594715271                                 | timestamp of bind expire  time                       | Yes          |
+
+#### For example
+
+* Mainnet
+```bash
+bnbcli bridge bind --symbol DEF-0F9 --amount     6000000000000000 --expire-time 1594715271 --contract-decimals 18 --from     alice --chain-id Binance-Chain-Nile --contract-address     0x6aade9709155a8386c63c1d2e5939525b960b4e7 --home ~/home_cli
```
Comment (HaoyangLiu) [inline docs/guides/concepts/bc-bridge.md:27]: remove redundent space

File: docs/guides/concepts/bc-bridge.md [hunk 27]
```diff
@@ -0,0 +1,130 @@
...
+| --from             | alice                                      | account name                                         | Yes          |
+| --channel-sequence | 1                                          | sequence of update bind  channel                     | Yes          |
+| --symbol           | DEF-0E9                                    | chain-id of the side  chain the validator belongs to | Yes          |
+| --contract-address | 0x6aade9709155a8386c63c1d2e5939525b960b4e7 | contract address of token  in smart chain            | Yes          |
+| --status           | success                                    | status of bind request                               | Yes          |
+
+#### For example**
+* Mainnet
+```bash
+bnbcli     bridge update-bind --symbol DEF-0F9 --from alice --chain-id     Binance-Chain-Nile --status success --contract-address      0x6aade9709155a8386c63c1d2e5939525b960b4e7 --channel-sequence     1 --home ~/home_cli
```
Comment (HaoyangLiu) [inline docs/guides/concepts/bc-bridge.md:46]: remove redundent space

File: docs/guides/concepts/bc-bridge.md [hunk 28]
```diff
@@ -0,0 +1,130 @@
...
+| --from             | alice                                      | account name                           | Yes          |
+| --to               | 0xf9f609f9f4309f191654aa1fd691a6be6aefa7ac | receiver address in smart  chain       | Yes          |
+| --amount           | 1000000000:BNB  (10 BNB)                   | amount of token to  transfer           | Yes          |
+| --expire-time      | 1594715271                                 | timestamp of transfer out  expire time | Yes          |
+
+#### For example
+
+* Mainnet
+```bash
+bnbcli bridge transfer-out --to     0xf9f609f9f4309f191654aa1fd691a6be6aefa7ac --expire-time 1594715271     --chain-id Binance-Chain-Nile --from alice --amount 100000000:DEF-0F9     --home ~/home_cli
```
Comment (HaoyangLiu) [inline docs/guides/concepts/bc-bridge.md:65]: remove redundent space

File: docs/guides/concepts/bc-bridge.md [hunk 29]
```diff
@@ -0,0 +1,130 @@
...
+| --channel-sequence | 1                                          | sequence of update  transfer out channel | Yes          |
+| --amount           | 1000000000:BNB  (10 BNB)                   | amount of token to  transfer             | Yes          |
+| --refund-reason    | timeout                                    | refund reason of transfer  out           | Yes          |
+| --refund-address   | bnb1tfh30c67mkzfz06as2hk0756mgdx8mgypu7ajl | refund address on binance  chain         | Yes          |
+
+#### For example
+* Mainnet
+```bash
+bnbcli     bridge transfer-out --to 0xf9f609f9f4309f191654aa1fd691a6be6aefa7ac     --expire-time 1594715271 --chain-id Binance-Chain-Nile --from alice     --amount 100000000:DEF-0F9 --home ~/home_cli
+```
```
Comment (HaoyangLiu) [inline docs/guides/concepts/bc-bridge.md:85]: remove redundent space

File: docs/faq/bsc/cross.md (hunk 4 above)
Comment (huangsuyu) [reply docs/faq/bsc/cross.md:7]: you can only send this kind of transaction with cli now

File: docs/guides/concepts/bc-bridge.md [hunk 30]
```diff
@@ -0,0 +1,130 @@
...
+| --symbol            | BNB                                                          | symbol of transfer token                                 | Yes          |
+| --relay-fee         | 1000:BNB                                                     | relay fee for transfer in  transaction                   | Yes          |
+| --expire-time       | 1594715271                                                   | expire time of transfer  in transaction on binance chain | Yes          |
+
+#### For example**
+
+* Mainnet
+```bash
+bnbcli     bridge transfer-in --channel-sequence 1 --contract-address      0x6aade9709155a8386c63c1d2e5939525b960b4e7 --refund-address     0xf9f609f9f4309f191654aa1fd691a6be6aefa7ac,0x052c89fa5c2f01082a63e91a29130edceca10806     --recipient-address  bnb1tfh30c67mkzfz06as2hk0756mgdx8mgypu7ajl,bnb13ls59kuvaaxq7a05rpps6z5r72nqcla3tuc4nc     --amount 1000,10000 --symbol BNB --relay-fee 1000:BNB --expire-time     1594715271 --chain-id Binance-Chain-Nile --from alice --home ~/home_cli
+```
```
Comment (HaoyangLiu) [inline docs/guides/concepts/bc-bridge.md:109]: ditto

File: docs/guides/concepts/bc-bridge.md [hunk 31]
```diff
@@ -0,0 +1,130 @@
...
+| --from               | alice             | account name                       | Yes          |
+| --channel-sequence   | 1                 | sequence of skip sequence  channel | Yes          |
+| --sequence-to-skip   | 1                 | sequence to skip                   | Yes          |
+| --claim-type-to-skip | 2                 | claim type to skip                 | Yes          |
+
+#### For example
+
+* Mainnet
+```bash
+bnbcli     bridge skip-sequence --channel-sequence 1 --sequence-to-skip 1     --claim-type-to-skip 1 --chain-id Binance-Chain-Nile --from alice --home     ~/home_cli
```
Comment (HaoyangLiu) [inline docs/guides/concepts/bc-bridge.md:128]: ditto

File: docs/faq/bsc/cross.md (hunk 4 above)
Comment (huangsuyu) [reply docs/faq/bsc/cross.md:7]: it's updated

File: docs/faq/bsc/cross.md (hunk 7 above)
Comment (huangsuyu) [reply docs/faq/bsc/cross.md:19]: it's corrected now

File: docs/faq/bsc/cross.md (hunk 8 above)
Comment (huangsuyu) [reply docs/faq/bsc/cross.md:17]: just added

File: docs/smart-chain/guides/concepts/consensus.md (hunk 15 above)
Comment (huangsuyu) [reply docs/smart-chain/guides/concepts/consensus.md:12]: fixed!

File: docs/faq/bsc/val.md (hunk 14 above)
Comment (huangsuyu) [reply docs/faq/bsc/val.md:35]: got it!

File: docs/smart-chain/guides/concepts/bsc-relayer.md (hunk 19 above)
Comment (huangsuyu) [reply docs/smart-chain/guides/concepts/bsc-relayer.md:13]: updated!

**Comment by edggge** [comment]
> *No description provided.*

😷😆

File: docs/smart-chain/validator/cross-chain-transfer.md [hunk 32]
```diff
@@ -0,0 +1,122 @@
...
+```bash
+tbnbcli bridge transfer-out --to 0xEe9546E92e6876EdF6a234eFFbD72d75360d91f0 --expire-time 1597543193 --chain-id Binance-Chain-Kongo --from owner --amount 100000000:BNB --node http://data-seed-prealpha-2-s1.binance.org:80
+```
+Result:
+```bash
+Committed at block 465899 (tx hash: 68FFF82197E27A3EC14AFF8C99A035FA9CA7120312AA55E98D11DFC0F8D9F3B9, response: {Code:0 Data:[] Log:Msg 0:  Info: GasWanted:0 GasUsed:0 Events:[{Type: Attributes:[{Key:[84 114 97 110 115 102 101 114 79 117 116 83 101 113 117 101 110 99 101] Value:[49 49] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0} {Key:[69 120 112 105 114 101 84 105 109 101] Value:[49 53 57 55 53 52 51 49 57 51] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0} {Key:[97 99 116 105 111 110] Value:[99 114 111 115 115 84 114 97 110 115 102 101 114 79 117 116] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}] Codespace: XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0})
+```
+
+## Transfer BNB from BSC to BC
+### Call **transferOut** of TokenHub contract in MyEtherWallet:
```
Comment (HaoyangLiu) [inline docs/smart-chain/validator/cross-chain-transfer.md:43]: > ### Call **transferOut** of TokenHub contract in MyEtherWallet:

Call **transferOut** of TokenHub contract in MyEtherWallet:

File: docs/smart-chain/validator/cross-chain-transfer.md [hunk 33]
```diff
@@ -0,0 +1,122 @@
...
+
+txValue = (amount + relayerFee)/1e18
+
+```
+
+After all the above parameters have been set to proper values, users can click the transact button to build transactions, and metamask plugin will be ejected. Then users can click the confirm button in metamask to sign and broadcast transactions.
+
+
+### batchTransferOut
+Call batchTransferOut of TokenHub contract in MyEtherWallet:
```
Comment (HaoyangLiu) [inline docs/smart-chain/validator/cross-chain-transfer.md:69]: > Call batchTransferOut of TokenHub contract in MyEtherWallet:

Call **batchTransferOut** of TokenHub contract in MyEtherWallet:

File: docs/smart-chain/validator/cross-chain-transfer.md [hunk 34]
```diff
@@ -0,0 +1,122 @@
...
+
+```
+
+## Transfer BEP2 to BSC
+Execute the following command to transfer ABC-A64 token to BSC:
+```bash
+tbnbcli bridge transfer-out --to 0xEe9546E92e6876EdF6a234eFFbD72d75360d91f0 --expire-time 1597543193 --chain-id Binance-Chain-Kongo --from owner --amount 10000000000:ABC-A64 --node http://data-seed-prealpha-2-s1.binance.org:80
+```
+## Transfer BEP2E to BC
+Before calling transferOut or batchTransferOut, users need to call approve method to grant enough allowance to TokenHub contract. For transferOut method, the allowance should equal the transfer amount. For batchTransferOut, the allowance should be the sum of the amount array.
```
Comment (HaoyangLiu) [inline docs/smart-chain/validator/cross-chain-transfer.md:102]: > Before calling transferOut or batchTransferOut, users need to call approve method to grant enough allowance to TokenHub contract. For transferOut method, the allowance should equal the transfer amount. For batchTransferOut, the allowance should be the sum of the amount array.

Before calling transferOut or batchTransferOut, users need to call **approve** method to grant enough allowance to TokenHub contract. For **transferOut** method, the allowance should greater than the transfer amount. For batchTransferOut, the allowance should be greater than the sum of the amount array.

File: docs/smart-chain/validator/cross-chain-transfer.md [hunk 35]
```diff
@@ -0,0 +1,122 @@
...
+| expireTime   | uint256 | Timestamp, counted by  second                                |
+| relayFee     | uint256 | The relayFee should be  no less than miniRelayFee. The initial miniRelayFee is 1e16. It can be  changed by  on-chain governance. |
+
+ The value here should be relayerFee/1e18. 
+
+
+
+### batchTransferOut
+
+![img](https://lh6.googleusercontent.com/kuEr0fCOv9kRtVZgEiyUtvKL_ixkuHKYy0J9hECGNWSdjNHpZKjKSioE6kLVK-WsGDl5xQvtZPR-YLJpJIw-z1P6qiwHXpMSYQ95esODXW8CzJgnUVbj-Qq1pgn4jKmoTX-kF1VS)
```
Comment (HaoyangLiu) [inline docs/smart-chain/validator/cross-chain-transfer.md:121]: Missing desciption for batchTransferOut: https://docs.google.com/document/d/1M-KCOkCs6pe2g2hFYuATTHAOiyln9zR2BPpBirx2wfc/edit#heading=h.j653uge95t4w

**Pull request merged**